core/include/vecmem/containers/impl/jagged_vector_buffer.ipp

```diff
@@ -141,37 +124,13 @@ jagged_vector_buffer<TYPE>::jagged_vector_buffer(
         ((host_access_resource != nullptr) ? m_outer_memory.get()
                                            : m_outer_host_memory.get());
 
-    // Size of the data payload.
-    const std::size_t dataSize =
-        std::accumulate(capacities.begin(), capacities.end(),
-                        static_cast<std::size_t>(0)) *
-        sizeof(TYPE);
-
-    //
-    // Construct a correctly aligned "start pointer" for the data elements.
-    //
-    // Construct the unaligned pointer by simply jumping over the "size array".
-    void* unaligned_start_ptr =
-        m_inner_memory.get() +
-        (capacities.size() * sizeof(typename value_type::size_type));
-    // The remaining size of the buffer, with the "size array" size removed.
-    std::size_t buffer_size =
-        m_inner_memory_size -
-        (capacities.size() * sizeof(typename value_type::size_type));
-    // Construct the "start pointer" using std::align.
-    char* start_ptr = static_cast<char*>(
-        std::align(alignof(TYPE), dataSize, unaligned_start_ptr, buffer_size));
-
     // Set up the vecmem::vector_view objects in the host accessible memory.
     std::ptrdiff_t ptrdiff = 0;
     for (std::size_t i = 0; i < capacities.size(); ++i) {
         new (host_ptr() + i) value_type(
             static_cast<typename value_type::size_type>(capacities[i]),
-            reinterpret_cast<typename value_type::size_pointer>(
-                m_inner_memory.get() +
-                i * sizeof(typename value_type::size_type)),
-            reinterpret_cast<TYPE*>(start_ptr + ptrdiff));
-        ptrdiff += capacities[i] * sizeof(TYPE);
+            &header_ptr[i], &data_ptr[ptrdiff]);
```

Comment:
The two are pretty much equivalent, but I think using `data_ptr + ptrdiff` instead of `&data_ptr[ptrdiff]` would be better here. That's why the variable is called `ptrdiff`, and is of type `std::ptrdiff_t` after all. :stuck_out_tongue: